maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java

```diff
@@ -86,6 +86,39 @@ public String getVersion()
         }
     }
 
+    @Override
+    public Artifact setVersion( String version )
+    {
+         String current = getVersion();
+         if ( current.equals( version ) || ( version == null && current.length() <= 0 ) )
```

Comment:
Could this be simplified?
```suggestion
         if ( current.equals( version ) || ( version == null && current.isEmpty() ) )
```

Reply:
While this is true, I wanted to retain the same code block as in `AbstractArtifact` for consistency reasons